payjoin/src/receive/optional_parameters.rs

```diff
@@ -23,6 +25,7 @@ impl Default for Params {
             disable_output_substitution: false,
             additional_fee_contribution: None,
             min_fee_rate: FeeRate::BROADCAST_MIN,
+            optimistic_merge: false,
```

Comment:
I believe `optimistic_merge` should be feature gated on `multi-party`, or a multi_party::Params that encapsulates receive::[v1?::]Params

Reply:
>or a multi_party::Params that encapsulates receive::[v1?::]Params
This would be my ideal solution. As we add new protocols on top of bip77 these params are going to more and more cluttered with no seperation for concerns. I'll move forward with a feature gate in order to keep the scope of this pr down and update with enapsulation in the future